### PR TITLE
Use supertest instead of repo-tools test app

### PR DESCRIPTION
**Pull request by fhinkel**
Trying to remove `repo-tools`. Will send a follow up PR that removes `repo-tools test deploy`. 

Ref: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/issues/1247

### CODE REVIEW

**Comment by fhinkel** [comment]
@bcoe Can you review this?

File: appengine/hello-world/standard/package.json [hunk 1]
```diff
@@ -15,15 +15,17 @@
   "scripts": {
     "deploy": "gcloud app deploy",
     "start": "node app.js",
-    "system-test": "repo-tools test app",
+    "system-test": "mocha --exit test/*.test.js",
```
Comment (bcoe) [inline appengine/hello-world/standard/package.json]: fun idea which you can take or leave, if you add `c8` as a dependency, it turns on Node's V8 coverage collection, so the user gets:

coverage

also understand if you think this is too much information for the user all at once.

File: appengine/hello-world/standard/package.json [hunk 2]
```diff
@@ -15,15 +15,17 @@
...
     "test": "npm run system-test",
     "e2e-test": "repo-tools test deploy"
   },
   "dependencies": {
     "express": "^4.16.3"
   },
   "devDependencies": {
-    "@google-cloud/nodejs-repo-tools": "^3.3.0"
+    "@google-cloud/nodejs-repo-tools": "^3.3.0",
```
Comment (bcoe) [inline appengine/hello-world/standard/package.json]: I think we can delete this dependency now?

Comment (fhinkel) [reply appengine/hello-world/standard/package.json]: No :) It's used in the e2e-test. But turns out, we're not calling 2e2-test from CI, instead we have special test files that run a deployment. But I'll prefer deleting it in a separate PR. There's a lot of magic happening in `repo-tools` based on env variables.

File: appengine/hello-world/standard/package.json (hunk 1 above)
Comment (fhinkel) [reply appengine/hello-world/standard/package.json]: Trying to keep dependencies as light as possible.

Comment (fhinkel) [reply appengine/hello-world/standard/package.json]: But lol, that coverage looks good!

Comment (JustinBeckwith) [inline appengine/hello-world/standard/package.json]: The `--exit` always makes me nervous, but it's not really blocking

Comment (fhinkel) [reply appengine/hello-world/standard/package.json]: These tests don't finish without `--exit`

Comment (bcoe) [reply appengine/hello-world/standard/package.json]: The default for `--exit` was changed relatively recently in mocha:

https://github.com/mochajs/mocha/pull/3034

I think it's reasonable to go back to the old default ... alternatively you'd need to capture the `server` object returned by `app.listen` and then call `.close()` on it when tests are complete... sounds like a hassle :stuck_out_tongue:

Comment (fhinkel) [reply appengine/hello-world/standard/package.json]: So by using `--exit` I'm just using Mocha the way it used to be, right?

**Pull request merged**